src/infra/vhost/virtualHostCmdRepo.go

```diff
@@ -2,125 +2,210 @@ package vhostInfra
+		"Hostname":        hostname,
+		"Aliases":         aliases,
+		"PublicDirectory": publicDir,
+		"CertPath":        infraData.GlobalConfigs.PkiConfDir + "/" + hostnameStr + ".crt",
+		"KeyPath":         infraData.GlobalConfigs.PkiConfDir + "/" + hostnameStr + ".key",
+		"MappingFilePath": mappingFilePath,
 	}
-	vhostFileStr := vhostFile.String()
 
-	hostnameStr := createDto.Hostname.String()
+	webServerConfigTemplate := `server {
+    listen 80;
+    listen 443 ssl;
+    server_name {{ .Hostname }} www.{{ .Hostname }}{{ range .Aliases }} {{ .String }} www.{{ .String }}{{ end }};
 
-	_, err = infraHelper.RunCmd(
-		"sed",
-		"-i",
-		`/server_name/ s/;$/ `+hostnameStr+` www.`+hostnameStr+`;/`,
-		vhostFileStr,
-	)
+    root {{ .PublicDirectory }};
+
+    ssl_certificate {{ .CertPath }};
+    ssl_certificate_key {{ .KeyPath }};
+
+    access_log /app/logs/nginx/{{ .Hostname }}_access.log combined buffer=512k flush=1m;
+    error_log /app/logs/nginx/{{ .Hostname }}_error.log warn;
+
+    include /etc/nginx/std.conf;
+    include {{ .MappingFilePath }};
+}`
+
+	webServerConfigTemplatePtr, err := template.
+		New("webServerConfigFile").
+		Parse(webServerConfigTemplate)
 	if err != nil {
-		return errors.New("CreateAliasFailed")
+		return "", errors.New("TemplateParsingError: " + err.Error())
 	}
 
-	// TODO: Regenerate cert for primary domain to include new alias
+	var webServerConfigFileContent strings.Builder
+	err = webServerConfigTemplatePtr.Execute(
+		&webServerConfigFileContent,
+		valuesToInterpolate,
+	)
+	if err != nil {
+		return "", errors.New("TemplateExecutionError: " + err.Error())
+	}
 
-	return infraHelper.ReloadWebServer()
+	return webServerConfigFileContent.String(), nil
 }
 
-func (repo VirtualHostCmdRepo) Create(createDto dto.CreateVirtualHost) error {
-	err := infraHelper.ValidateWebServerConfig()
+func (repo *VirtualHostCmdRepo) createWebServerFile(
+	publicDir valueObject.UnixFilePath,
+	hostname valueObject.Fqdn,
+) error {
+	aliases, err := repo.queryRepo.ReadAliasesByHostname(hostname)
 	if err != nil {
-		return err
+		return errors.New("GetAliasesByHostnameError: " + err.Error())
 	}
 
-	hostnameStr := createDto.Hostname.String()
-
-	if createDto.Type.String() == "alias" {
-		return repo.createAlias(createDto)
+	aliasesHostnames := []valueObject.Fqdn{}
+	for _, alias := range aliases {
+		aliasesHostnames = append(aliasesHostnames, alias.Hostname)
 	}
 
-	publicDir := infraData.GlobalConfigs.PrimaryPublicDir + "/" + hostnameStr
-	certPath := infraData.GlobalConfigs.PkiConfDir + "/" + hostnameStr + ".crt"
-	keyPath := infraData.GlobalConfigs.PkiConfDir + "/" + hostnameStr + ".key"
-	mappingFilePath := "/app/conf/nginx/mapping/" + hostnameStr + ".conf"
-
-	nginxConf := `server {
-    listen 80;
-    listen 443 ssl;
-    server_name ` + hostnameStr + ` www.` + hostnameStr + `;
-
-    root ` + publicDir + `;
+	hostnameStr := hostname.String()
+	if infraHelper.IsPrimaryVirtualHost(hostname) {
+		hostnameStr = "primary"
+	}
```

Comment:
What, hostnameStr vira a palavra "primary", why?

Reply:
Troquei o nome da variável para "vhostHostnameConfFileName" e adicionei o ".conf" no final, acho que faltou o nome correto ali pra poder entender que esse é o nome do arquivo e não do hostname em si.
De qualquer forma, nós temos, por padrão, que o nome dos arquivos do VirtualHost primário é chamado de "primary" (`primary.conf`), especialmente no nome de seus arquivos, portanto eu faço a validação na linha 102 para descobrir se é o nosso vhost primário, caso for, dou o nome de arquivo correto pra ele: "primary".